version: "3"

services:
  nginx: 
    image: mapic/shiny-nginx:latest
    env_file: config/nginx.env
    restart: unless-stopped
    volumes:
      - ./config/nginx.conf:/etc/nginx/nginx.conf
      - ./config/nginx.certbot.conf:/etc/nginx/nginx.certbot.conf
      - ./ssl/:/home/ssl/
    ports:
      - "80:80"
      - "443:443"
    logging:
      options:
        max-size: "5k"
        max-file: "10"
    command: "bash docker-entrypoint.sh"

  auth0:
    image: mapic/shiny-auth0:latest
    env_file: config/auth0.env
    command: "npm start"
    restart: unless-stopped
    # volumes:                              # for debug only
      # - ../shiny-auth0/app/:/usr/src/app  # for debug only
    # command: "bash docker-entrypoint.sh"  # for debug only

  shiny:
    image: mapic/shiny-server-pro:latest
    env_file: config/shiny.env
    restart: unless-stopped
    logging:
      options:
        max-size: "20k"
        max-file: "10"
    volumes:
      - ./config/shiny-server.conf:/etc/shiny-server/shiny-server.conf
      - shiny-data:/srv/shiny-server/apps # shiny data volume shared with sftp folder

  sftp:
    image: atmoz/sftp
    restart: unless-stopped
    volumes:
        - shiny-data:/home/shiny/upload # shiny data volumne shared with shiny
        - ./config/sftp.users.conf:/etc/sftp/users.conf:ro
    ports:
        - "2222:22"

networks:
  default:
    external:
      name: shiny-network  

volumes:
  shiny-data: